abstract_domains/lattice.py

```diff
@@ -164,35 +164,67 @@ def replace(self, other: 'Lattice') -> 'Lattice':
         return self
 
 
-class BoundedLattice(Lattice, metaclass=ABCMeta):
-    """A mutable lattice element, with default bottom and top elements."""
+class Kind(Enum):
+    TOP = 3
+    DEFAULT = 2
+    BOTTOM = 1
+
 
-    class Kind(Enum):
-        TOP = 3
-        DEFAULT = 2
-        BOTTOM = 1
+class KindMixin(Lattice, metaclass=ABCMeta):
```

Comment:
It is needed whenever a class does intentionally not define all abstract methods